ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -334,7 +327,9 @@ def steps_from_proto(
                             for k in range(a_size)
                         ]
             action_mask = (1 - mask_matrix).astype(np.bool)
-            indices = _generate_split_indices(behavior_spec.discrete_action_branches)
+            indices = _generate_split_indices(
+                behavior_spec.action_spec.discrete_action_branches
```

Comment:
I think there are too many repeats in this line. I think the API needs to change. This could be : 
```suggestion
                behavior_spec.actions.discrete_branches
```